pkg/controller/controller.go

```diff
@@ -140,6 +140,10 @@ const (
 	snapshotNotBound = "snapshot %s not bound"
 
 	pvcCloneFinalizer = "provisioner.storage.kubernetes.io/cloning-protection"
+
+	// The kv that needs to be automatically inserted into CreateVolumeRequest.Parameters
+	parameterPrefix = "X-K8S-PROVISIONER-"
+	ParameterScName = parameterPrefix + "SC-NAME"
```

Comment:
if you want the SC name to be passed to the CSI driver, add one more parameter to the storageclass `kubernetes-sc-name: sc-name` and it will get automatically passed to the CSI driver?

Reply:
So there are parameters you can pass to the SC, and those parameters are passed to the CSI provisioner. I think that would solve your issue?

Reply:
> if you want the SC name to be passed to the CSI driver, add one more parameter to the storageclass `kubernetes-sc-name: sc-name` and it will get automatically passed to the CSI driver?

in perpareProvision function.

Reply:
> So there are parameters you can pass to the SC, and those parameters are passed to the CSI provisioner. I think that would solve your issue?

yes，prepareProvision has sc information，the parameters are passed to the sci，so I define a constant key here and add sc name to the parameters behind.

Reply:
> So there are parameters you can pass to the SC, and those parameters are passed to the CSI provisioner. I think that would solve your issue?

I got it, you mean I add it to sc when created, I did that before, the problem is `ParameterScName`  is required,  If it does not exist, I am expecting an error when creating，because delayed error messages are not user friendly, and users will be confused about this parameter, so I did it in external-provisioner.

Reply:
yap I think so. But many coworkers didn't think so. In the end, we maintain our external provisioner forked here.
Thanks, It can be closed.